### PR TITLE
Fix filters list/prune in image http compat/libpod api endpoints

### PR DESCRIPTION
**Pull request by jmguzik**
Signed-off-by: Jakub Guzik <jakubmguzik@gmail.com>

/kind bug
As described in previous commits and PRs, in certain situations prune/list filter handling is broken (see #9773, #9711 and #9758 ). The problem affects images as well. This is an attempt to fix it.

Adding also handling of dangling filter in compat API (was unimplemented).

### CODE REVIEW

File: pkg/api/handlers/compat/images_prune.go [hunk 1]
```diff
@@ -8,36 +8,29 @@ import (
...
 
 func PruneImages(w http.ResponseWriter, r *http.Request) {
 	var (
 		filters []string
 	)
-	decoder := r.Context().Value("decoder").(*schema.Decoder)
 	runtime := r.Context().Value("runtime").(*libpod.Runtime)
 
-	query := struct {
-		All     bool
```
Comment (jmguzik) [inline pkg/api/handlers/compat/images_prune.go:24]: I am not really sure what this one is doing here. I feel it does nothing. I know that All is used in libpod API, but compat has no such thing. As in my opinion, it is a dead code, I decided to remove it. Please let me know what you think.

Comment (rhatdan) [reply pkg/api/handlers/compat/images_prune.go:24]: SGTM

**Comment by rhatdan** [comment]
/approve
LGTM
@containers/podman-maintainers PTAL

**Comment by TomSweeneyRedHat** [comment]
LGTM
@mheon want to LGTM and merge?

**Pull request merged**